onnxruntime/core/graph/function.cc

```diff
@@ -29,6 +149,13 @@ void IOTypeConstraintHelper(const ONNX_NAMESPACE::FunctionProto& onnx_func_proto
   std::vector<std::pair<std::string, std::string>> output_types_list(onnx_func_proto_.output_size());
   std::unordered_map<std::string, std::vector<std::string>> type_constraint_map;
   std::unordered_map<std::string, ONNX_NAMESPACE::AttributeProto_AttributeType> attribute_type_map;
+
+  // Create an all permissive list of data types. This will be used in case of model local functions
+  // when we cannot infer the type constraints from function proto body
+  std::unordered_set<std::string> all_types;
+  all_types.insert(ONNX_NAMESPACE::OpSchema::all_tensor_types_with_bfloat().cbegin(), ONNX_NAMESPACE::OpSchema::all_tensor_types_with_bfloat().cend());
+  all_types.insert(ONNX_NAMESPACE::OpSchema::all_tensor_sequence_types().cbegin(), ONNX_NAMESPACE::OpSchema::all_tensor_sequence_types().cend());
+
```

Comment:
nit: line length.

[https://marketplace.visualstudio.com/items?itemName=PaulHarrington.EditorGuidelines](https://marketplace.visualstudio.com/items?itemName=PaulHarrington.EditorGuidelines) is helpful as you can set a vertical line at 120 chars in the editor in VS. 

Can also do the same in VS Code with this setting
    "editor.rulers": [    
        120
    ],